homeassistant/components/tesla_fleet/application_credentials.py

```diff
@@ -23,15 +25,16 @@ async def async_get_auth_implementation(
     return TeslaOAuth2Implementation(
         hass,
         DOMAIN,
```

Comment:
Since we're always passing the integration domain there will only ever be one implementation instance.

When using application credentials we're meant to use the auth domain for the implementation so that each credentials instance will generate one implementation instance.

Reply:
Martin I have tried to address your feedback and fix the implementations, and also add support for custom implementations in https://github.com/home-assistant/core/pull/123324